src/utils/CoreAPI/index.ts

```diff
@@ -0,0 +1,164 @@
+   * @param password Password to login with
+   *
+   * @return {boolean} Success status (true if successful)
+   */
+  public static async LOGIN(username: string, password: string): Promise<{ uuid: string, jwt: string }> {
+    const url = `${process.env.CORE_HOST}:${process.env.CORE_PORT}${process.env.LOGIN_PATH}`;
+    const method: CoreRequestMethod = 'POST';
+    const headers = new Headers({
+      'content-type': 'application/json',
+    });
+
+    const body = {
+      username,
+      password,
+    };
+
+    const request = new Request(url, {
+      method,
+      headers,
+      body: JSON.stringify(body),
+    });
+
+    const response = await fetch(request);
+
+    // Check response
+    if (!response.ok) throw new AuthenticationError();
+
+    const { token: jwt, uuid } = await response.json();
+    localStorage.setItem('token', jwt);
+
+    return { uuid, jwt };
+  }
+
+  /**
+   * Get a row from the core
+   *
+   * @param uuid UUID of the desired row
+   *
+   * @returns
+   */
+  public async get<T>(uuid: string): Promise<CoreRow<T>> {
+    const url = `${this.url}/${uuid}`;
+    const method: CoreRequestMethod = 'GET';
+    const request = new Request(url, this.getOptions(method));
+
+    return await this.coreFetch(request) as CoreRow<T>;
+  }
+
+  /**
+   * Get all of the specified resource
+   */
+  public async getAll<T>(): Promise<CoreRow<T>[]> {
+    const url = this.url;
+    const method: CoreRequestMethod = 'GET';
+    const request = new Request(url, this.getOptions(method));
+
+    return this.coreFetch(request);
+  }
+
+  /**
+   * Update a row on the core
+   *
+   * @param row Information to overwrite
+   */
+  public async update<T>(row: CoreUpdateRequest<T>): Promise<CoreRow<T>> {
+    const url = this.url;
+    const method: CoreRequestMethod = 'PUT';
+    const request = new Request(url, this.getOptions(method, row));
+
+    return this.coreFetch(request);
+  }
+
+  /**
+   * Create a row in the database
+   *
+   * @param row Row to create
+   */
+  public async create<T>(row: CoreCreationRequest<T>) {
+    const url = this.url;
+    const method: CoreRequestMethod = 'POST';
+    const request = new Request(url, this.getOptions(method, row));
+
+    return this.coreFetch(request);
+  }
+
+  /**
+   * Send a HEAD request for the resources
+   *
+   * @param timestamp Timestamp to compare
+   */
+  public async head(timestamp: string) {
+
+  }
+
+  private getOptions<T>(method: CoreRequestMethod, body?: T): RequestInit {
+
+    const jwt = localStorage.getItem('token');
+    if (!jwt) {
+      throw new Error('Not authenticated');
+    }
+    const headers = new Headers({
+      'content-type': 'application/json',
+      Authorization: `Bearer ${jwt}`,
+    });
+
+    return {
+      method,
+      headers,
+      mode: 'cors',
+      ...body && { body: JSON.stringify(body) },
+    };
+  }
+
+  private async coreFetch(request: Request) {
+    // Declare block scoped variables (let) at top of block
```

Comment:
would it be a good idea to make the request here? so that every request gets the options?

Reply:
Not all the options are going to be the same for every request, such as login